google_guest_agent/network/manager/systemd_networkd_linux.go

```diff
@@ -0,0 +1,156 @@
+	"github.com/GoogleCloudPlatform/guest-agent/metadata"
+	"github.com/GoogleCloudPlatform/guest-logging-go/logger"
+)
+
+type systemdNetworkd struct {
+	ConfigDir string
+}
+
+// init adds this network manager service to the list of known network managers.
+func init() {
+	registerManager(systemdNetworkd{
+		ConfigDir: "/usr/lib/systemd/network",
+	})
+}
+
+// Name returns the name of the network manager service.
+func (n systemdNetworkd) Name() string {
+	return "systemd-networkd"
+}
+
+// IsManaging checks whether systemd-networkd is managing the provided interface.
+// This first checks if the systemd-networkd service is running, then uses networkctl
+// to check if systemd-networkd is managing or has configured the provided interface.
+func (n systemdNetworkd) IsManaging(ctx context.Context, iface string) (bool, error) {
+	// First check if the service is running.
+	res := run.WithOutput(ctx, "systemctl", "is-active", "systemd-networkd.service")
+	if res.ExitCode != 0 {
+		return false, nil
+	}
+
+	// Check systemd network configuration.
+	res = run.WithOutput(ctx, "/bin/sh", "-c", fmt.Sprintf("networkctl status %s --json=short", iface))
+	if res.ExitCode != 0 {
+		return false, fmt.Errorf("failed to check systemd-networkd network status: %v", res.StdErr)
+	}
+
+	interfaceStatus := make(map[string]any)
+
+	if err := json.Unmarshal([]byte(res.StdOut), &interfaceStatus); err != nil {
+		return false, fmt.Errorf("failed to unmarshal interface status: %v", err)
+	}
+
+	administrativeState, found := interfaceStatus["AdministrativeState"]
+	if !found {
+		return false, fmt.Errorf("could not determine interface state, AdministrativeState was not present")
+	}
+
+	return administrativeState == "configured", nil
+}
+
+// Setup sets up the non-primary network interfaces for systemd-networkd by writing
+// configuration files to /usr/lib/systemd/network.
+func (n systemdNetworkd) Setup(ctx context.Context, config *cfg.Sections, payload []metadata.NetworkInterfaces) error {
+	// Create a network configuration file with default configurations for each network interface.
+	googleInterfaces, googleIpv6Interfaces := InterfaceListsIpv4Ipv6(payload)
+	if err := configureIPv6(ctx, googleIpv6Interfaces, payload[0]); err != nil {
+		return fmt.Errorf("error configuring IPv6: %v", err)
+	}
+	for _, iface := range googleInterfaces {
+		logger.Debugf("write %s network config for %s", n.Name(), iface)
+
+		var config *os.File
+
+		var dhcp = "ipv4"
+		if slices.Contains(googleIpv6Interfaces, iface) {
+			dhcp = "yes"
+		}
+
+		// The number in beginning of the file name signifies its priority. We want this
+		// to take priority before any of the default configurations that systemd-networkd
+		// has set.
+		config, err := os.Create(fmt.Sprintf("%s/50-%s-google-guest-agent.network", n.ConfigDir, iface))
+		if err != nil {
+			return err
+		}
+		defer config.Close()
+
+		// Default, simple configuration for any network interface.
+		contents := []string{
```

Comment:
this is a ini file, you can use `github.com/go-ini/ini` (see comment bellow for unmarshalling too).

Reply:
Done.